pkg/users/account-create-dialog.js

```diff
@@ -42,6 +42,16 @@ function AccountCreateBody({ state, errors, change }) {
         locked, change_passw_force
     } = state;
 
+    // We want to let user know that password and confirmation password do not match without them having to constantly click on "Create" to validate the form.
+    // But we also don't want to show an error message while they are typing the confirmation password, only when they are finished typing it.
+    // To solve the issue of telling that user is finished writing the confirmation password, let's do the following:
+    // Since both passwords should be the same length, we should show error message while user is typing,
+    // but only if the confirm password is same length (or higher) than the original password (i.e. they are finished typing)
+    let dynamic_password_confirm_error;
+    if (state.password_confirm.length >= state.password.length)
+        dynamic_password_confirm_error = validate_password_confirm(state.password_confirm, state.password);
```

Comment:
I would let @garrett to decide what we want here but does this mean that we don't show error if the passwords are different length? Which I think is the most usual case when passwords differ that you hit by accident one more key or forget some part of it. Or does this also validate after you blur? Which I think should be the primary method for validating fields.

Reply:
Also this is validating one field under very specific circumstances. Why not validating all fields on blur?

Reply:
I agree onBlur is also a nice alternative, which I would like to see used more not only in this form but across the whole cockpit. 
But redesigning the whole form to work with onBlur is a bit bigger, since you have to redo the whole additional login related to "Create with weak password", etc. 
And I wanted to keep this PR small, since this PR is blocking #18470  and #18353 , I want to get it in quickly, so I saw this approach a nice fix without having to redo the whole validation for the whole dialog.

But again, I'm not against onBlur, I like that approach too, so if we agree to use it, in that case I would split the validation refactoring into a separate PR, so this can get merged quickly and not block the other 2 PRs.

Reply:
I thought I replied to this yesterday. I don't see it here though. (Or perhaps I'm thinking of a similar conversation?)

We should try to catch things like this on the fly, before submitting them. But I'm fine if that's a follow-up.

Related: @jelly and I were talking about improving validation in Cockpit. He was looking into it. It's probably worth having a larger conversation about this across Cockpit and improving everything, everywhere. (I was also talking with the PatternFly team about validation too. It's an odd omission that they just do not cover.) It's not necessarily related to this PR, but having guidelines and examples in place would make it clearer for PRs like this in the future. (And perhaps @skobyda could talk with @jelly if there's a good technique for this, even for this PR or a follow-up perhaps.)

But there are so many other changes here, I'm OK with improving this specifically in a follow-up.

Reply:
Solution discussed on Matrix:
```
Garrett
skobyda: if you're typing, then you're not done. but if if you trigger the check and they don't match, then you should be told that until they do match.
Garrett
skobyda: the delayed check is just to cover the part where you're typing and not done
Garrett
skobyda: but it shouldn't check and then not check and go back and forth on checking... especially when you trigger it and then they're obviously not matching (different lengths)

....

skobyda
Garrett: sooo, I should wait until both passwords are the same length, and if they are and they don't match, only then start checking after each keystroke?
Garrett
this would not be a problem if we did away with the password check and had a toggle visibility... ahem
Garrett
but anyway, yes
```

So not this PR implements it this way:

The dialog does not validate if passwords match until:
1. they are at least the same length (which signals user has finished typing)
OR
2. user clicks on "Create" (which also signals they are finished typing)
Once that happens, and passwords do not match, the confirm password is validated after each keystroke.

This still keeps the old form onSubmit validation, which @marusak was concerned would be lost, and also adds the new approach which tells user more dynamically if the passwords do not mathc.